Only allow students to view scores or report if option is allowed.

peterdietz opened the pull request:
If options such as viewReport or viewScore were unchecked, a student would still be able to see them. This fixes that, and you now only see things when they are allowed.

![1je5uw](https://cloud.githubusercontent.com/assets/58014/22844339/c492ccd0-efab-11e6-92e1-30142e4d300c.gif)

baholladay commented on lib.php:
```diff
@@ -123,16 +123,43 @@ public function get_links($linkarray) {
             return $results['error'];
         }
+
+        global $DB;
```
Any global declarations should be at the top of the function

baholladay commented on lib.php:
```diff
@@ -123,16 +123,43 @@ public function get_links($linkarray) {
         }
+
+        global $DB;
+        $plagiarismvalues = $DB->get_records_menu('plagiarism_vericite_config', array('cm' => $vericite['cmid']), '', 'name,value');
```
This is terribly inefficient. This function gets calls for every single submission, so in a class of 100's of students, you'd be looking up the same record 100's of times. Is there a different place to put this logic or get this setting?

baholladay commented on lib.php:
```diff
@@ -123,16 +123,43 @@ public function get_links($linkarray) {
+        $viewsimilarityscore = false;
+        $viewfullreport = false;
+
+        if ($plagiarismvalues['plagiarism_show_student_score'] == 1) {
```
where do you check if the user is a student or instructor?

peterdietz commented:
Pushed the squashed commit up, that has the instructor info

baholladay commented on lib.php:
```diff
@@ -123,16 +123,31 @@ public function get_links($linkarray) {
         } else {
             // User only sees similarity score
-            $output = '<span class="vericite-report">' . get_string('similarity', 'plagiarism_vericite') . $similaritystring . '</span>';
+            if($results['viewSimilarityScore']) {
```
should this also include the isInstructor check?

baholladay replied:
also, merge the }else{ into an }else if{

peterdietz committed:
Only allow students to view scores or report if option is allowed.

Not make extra query to get canView data for report display

Add isInstructor to results array

peterdietz commented:
Combined the else .. if to an else if. And added the instructor check.
